happi/client.py

```diff
@@ -801,10 +801,32 @@ def from_config(cls, cfg=None):
         cfg_parser = configparser.ConfigParser()
         cfg_file = cfg_parser.read(cfg)
         logger.debug("Loading configuration file at %r", cfg_file)
-        db_kwargs = cfg_parser['DEFAULT']
+        if len(cfg_parser.keys()) > 1:
+            # multi backend configuration have multiple sections
+            multi_backend = BACKENDS['multi']
+            sub_backends = []
+            for section in cfg_parser.keys():
+                bknd = cls._get_backend_from_config(cfg_parser[section], cfg)
+                sub_backends.append(bknd)
+            # move DEFAULT from index 0 to -1
+            default_backend = sub_backends.pop(0)
```

Comment:
It's a bit arbitrary to have DEFAULT go to the end, as I think a similar argument could be made for it to be highest priority.

I think the as-defined order makes more sense and would vote to remove this line

Reply:
My reasoning here was that smaller backends could essentially patch the DEFAULT backend.  In many cases the additional devices might not be unique, but slightly modified versions of devices existing in the DEFAULT backend.  In that case if DEFAULT was highest priority, the client wouldn't return the patched version.

Python's `configparser` also always pulls DEFAULT to be the first section listed, regardless of its placement in the cfg file itself, so that aspect of the ordering is lost.  Lacking a better way of reclaiming that order I thought it'd make sense to toss it at the end

Reply:
I guess I'm not sure why we'd have backends with the same (or very similar) information in them. I'd expect it to be used like "backend 1 has the hutch happi database" and "backend 2 has the photon side happi database" and "backend 3 has the accelerator side happi database", so priority wise search "hutch -> photon -> accelerator". 

Interesting about `DEFAULT` always coming first. `DEFAULT` coming first in configparser and last in happi feels weird, but perhaps it's just me. In that case, I'd say for the multi-backend scenario it should be recommended to _not_ use `DEFAULT` at all so you have clear control over the ordering.

Reply:
> Python's configparser also always pulls DEFAULT to be the first section listed, regardless of its placement in the cfg file itself, so that aspect of the ordering is lost.

I didn't know this. I guess we shouldn't be using DEFAULT at all in these config files then, to preserve some sort of logical order?

I think by and large patching over the main db isn't the main use case of this, rather I'd expect the different dbs to be responsible for different subsets of the beamline (mostly)

Reply:
> I guess I'm not sure why we'd have backends with the same (or very similar) information in them. 

The use case I had in mind was for something like XPP, where they can move devices between branches.  The information would be identical except for a few fields, so instead of changing the device_config they could just add a patching config. (if I understood the use case right)

> DEFAULT coming first in configparser and last in happi feels weird, but perhaps it's just me. 

Though the ordering is different, the behavior actually matches I think.  In both cases, fields in non-`DEFAULT` configs take precedence, and if there is a missing entry in one of the non-`DEFAULT` sections, the value from `DEFAULT` is used.  Phrase another way, `DEFAULT` never overrides information from a different section.  (the ordering of keys in the `ConfigParser` doesn't actually matter for the config files itself`)

Reply:
I guess this gives us a lot of flexibility over how to do mode swaps: shadowing, holding two configs that get swapped, etc.

Reply:
I think we can still use DEFAULT, but not using it would probably make the config priority clearer for users.

Reply:
A quick update to the config parsing logic to make the DEFAULT section optional